src/ui/steps.py

```
@@ -4,59 +4,121 @@
+
+    Returns:
+        tuple[bool, str]: Success flag and content-based signature.
+    """
+    if uploaded is None:
+        return False, ""
+    try:
+        suffix = pathlib.Path(uploaded.name).suffix.lower().lstrip(".")
+        if suffix == "csv":
+            df_new = pd.read_csv(uploaded)
+        else:
+            df_new = pd.read_excel(uploaded)
```

Ruff (0.15.12)

[warning] Use ternary operator `df_new = pd.read_csv(uploaded) if suffix == "csv" else pd.read_excel(uploaded)` instead of `if`-`else`-block

Replace `if`-`else`-block with `df_new = pd.read_csv(uploaded) if suffix == "csv" else pd.read_excel(uploaded)`

(SIM108)